contentreview-impl/impl/src/test/org/sakaiproject/contentreview/logic/TurnitinImplTest.java

```diff
@@ -30,5 +88,195 @@ public void testFileEscape() {
+		expect(user.getEmail()).andStubReturn("email@mail.es");
+		replay(user);
+		expect(M_use.getUser("userId")).andStubReturn(user);
+		tiiService.setUserDirectoryService(M_use);
+		replay(M_use);
+		
+		M_per = createMock(SakaiPersonManager.class);
+		tiiService.setSakaiPersonManager(M_per);
+		SakaiPerson person = createMock(SakaiPerson.class);
+		expect(M_per.getSystemMutableType()).andStubReturn(null);
+		expect(M_per.getSakaiPerson("userId",null)).andStubReturn(person);
+		replay(M_per);
+		
+		M_ss = createMock(SiteService.class);
+		tiiService.setSiteService(M_ss);
+		Site siteA = new FakeSite("siteId");
+		expect(M_ss.getSite("siteId")).andStubReturn(siteA);
+		replay(M_ss);
+		
+		M_con = createMock(ContentHostingService.class);
+		tiiService.setContentHostingService(M_con);
+		ContentResource r = createMock(ContentResource.class);
+		ResourceProperties rp = createMock(ResourceProperties.class);
+		expect(M_con.getResource("contentId")).andStubReturn(r);
+		expect(r.getProperties()).andStubReturn(rp);
+		expect(r.getId()).andStubReturn("contentId");
+		expect(rp.getNamePropDisplayName()).andStubReturn("displayName");
+		expect(rp.getProperty("displayName")).andStubReturn("fileName");
+		expect(rp.getProperty("turnitin_id")).andStubReturn("123456");
+		replay(M_con);
+		replay(r);
+		replay(rp);
+		
+		FakeSiteAdvisor siteAdvisor = new FakeSiteAdvisor();
+		tiiService.setSiteAdvisor(siteAdvisor);
+		
+		M_assi = createMock(AssignmentService.class);
+		tiiService.setAssignmentService(M_assi);
+		org.sakaiproject.assignment.api.Assignment assignA = createMock(org.sakaiproject.assignment.api.Assignment.class);
+		expect(M_assi.getAssignment("taskId")).andStubReturn(assignA);
+		AssignmentContent contentA = createMock(AssignmentContent.class);
+		expect(assignA.getContent()).andStubReturn(contentA);
+		ResourceProperties rp2 = createMock(ResourceProperties.class);
+		expect(contentA.getProperties()).andStubReturn(rp2);
+		expect(rp2.getProperty("turnitin_id")).andStubReturn("123456");
+		replay(M_assi);
+		replay(assignA);
+		replay(contentA);
+		replay(rp2);
+		
+		FakeTiiUtil M_util = new FakeTiiUtil();
+		tiiService.setTiiUtil(M_util);
+		
+		TurnitinAccountConnection tac = new TurnitinAccountConnection();
+		tac.setUseSourceParameter(false);
+		tiiService.setTurnitinConn(tac);
+		
+		M_dao = (ContentReviewDao)applicationContext.getBean("org.sakaiproject.contentreview.dao.ContentReviewDao");
+		if (M_dao == null) {
+			throw new NullPointerException("DAO could not be retrieved from spring context");
+		}
+
+		tiiService.setDao(M_dao);
+
+		/*ContentReviewItem item = new ContentReviewItem("userId", "siteId", "taskId", "contentId", new Date(), ContentReviewItem.NOT_SUBMITTED_CODE);
+		item.setNextRetryTime(new Date());
+		item.setUrlAccessed(false);
+		item.setSubmissionId("submissionId");*/
+		
+		//TEST 1
+		tiiService.queueContent("userId", "siteId", "taskId", "contentId", "submissionId");	
```

Comment:
Problem is all tiiService methods I'm checking are void methods, not sure what to "assert". I added all the logic needed for every method to get to the LTI call and process the returned value. Obviously the LTI calls result is always positively faked.